snapcraft/internal/repo/_deb.py

```diff
@@ -270,6 +270,24 @@ def get_package_libraries(cls, package_name):
 
         return _library_list[package_name].copy()
 
+    @classmethod
+    def get_package_for_file(cls, file_path: str) -> str:
+        try:
+            output = (
+                subprocess.check_output(
+                    ["dpkg", "-S", file_path], stderr=subprocess.STDOUT
```

Comment:
if there is a case where this may be called without the leading '/' (i.e. relative instead of absolute path), the -S will effectively wildcard the search term potentially yielding undesired results.  may be safest to add leading '/' if not present?

Reply:
Though i see now that's how you invoked it.... maybe move the join(os.sep, file_path) into here?

Reply:
Good idea, it is idempotent to do that so better to do it always